app/services/core/rx-storage-list-proxy.ts

```diff
@@ -0,0 +1,87 @@
+import { Type } from "@angular/core";
+import { Observable } from "rxjs";
+
+import { ServerError } from "app/models";
+import { exists } from "app/utils";
+import { StorageClientService } from "../storage-client.service";
+import { CachedKeyList } from "./query-cache";
+import { RxListProxy, RxListProxyConfig } from "./rx-list-proxy";
+
+const defaultOptions = {
+    maxResults: 50,
+};
+
+export interface RxStorageListProxyConfig<TParams, TEntity> extends RxListProxyConfig<TParams, TEntity> {
+    proxyConstructor: (client: any, params: TParams, options: any) => any;
+    initialOptions?: any;
+}
+
+/**
+ * List proxy implementation to use the storage client proxy
+ */
+export class RxStorageListProxy<TParams, TEntity> extends RxListProxy<TParams, TEntity> {
+    private _nextLink = null;
+    private _loadedFirst = false;
+    private _proxyConstructor: (client: any, params: TParams, options: any) => any;
+
+    constructor(
+        type: Type<TEntity>,
+        private storageClient: StorageClientService,
+        config: RxStorageListProxyConfig<TParams, TEntity>) {
+
+        super(type, config);
+        this._proxyConstructor = config.proxyConstructor;
```

Comment:
Would rename that to be something else as you are not creating the list proxy here. Maybe `getData`